runtime/parachains/src/paras_inherent.rs

```diff
@@ -24,30 +24,106 @@
 use frame_support::{
+	fail,
 	inherent::{InherentData, InherentIdentifier, MakeFatalError, ProvideInherent},
 	pallet_prelude::*,
+	traits::Randomness,
 };
 use frame_system::pallet_prelude::*;
+use pallet_babe::CurrentBlockRandomness;
 use primitives::v1::{
-	BackedCandidate, InherentData as ParachainsInherentData, ScrapedOnChainVotes,
-	PARACHAINS_INHERENT_IDENTIFIER,
+	BackedCandidate, CandidateHash, CoreIndex, InherentData as ParachainsInherentData,
+	ScrapedOnChainVotes, SessionIndex, SigningContext, UncheckedSignedAvailabilityBitfield,
+	UncheckedSignedAvailabilityBitfields, ValidatorId, PARACHAINS_INHERENT_IDENTIFIER,
 };
+use rand::Rng;
+use scale_info::TypeInfo;
 use sp_runtime::traits::Header as HeaderT;
-use sp_std::prelude::*;
+use sp_std::{
+	collections::{btree_map::BTreeMap, btree_set::BTreeSet},
+	prelude::*,
+};
 
 pub use pallet::*;
 
 const LOG_TARGET: &str = "runtime::inclusion-inherent";
 // In the future, we should benchmark these consts; these are all untested assumptions for now.
-const BACKED_CANDIDATE_WEIGHT: Weight = 100_000;
 const INCLUSION_INHERENT_CLAIMED_WEIGHT: Weight = 1_000_000_000;
 // we assume that 75% of an paras inherent's weight is used processing backed candidates
 const MINIMAL_INCLUSION_INHERENT_WEIGHT: Weight = INCLUSION_INHERENT_CLAIMED_WEIGHT / 4;
 
+/// Weights
+///
+/// TODO replaced this with the true weights
+/// TODO as part of <https://github.com/paritytech/polkadot/pull/4044>
+/// TODO The current ones are _preliminary_ copies of those.
+fn paras_inherent_total_weight<T: Config>(
+	n_backed_candidates: u32,
+	n_bitfields: u32,
+	n_disputes: u32,
+) -> Weight {
+	let weights = T::DbWeight::get();
+	// static
+	(10_901_789_000 as Weight)
+		// backed candidates
+		.saturating_add((424_633_000 as Weight).saturating_mul(n_backed_candidates as Weight))
+		.saturating_add(weights.reads(58 as Weight))
+		.saturating_add(weights.reads((8 as Weight).saturating_mul(n_backed_candidates as Weight)))
+		.saturating_add(weights.writes(212 as Weight))
+		.saturating_add(weights.writes((2 as Weight).saturating_mul(n_backed_candidates as Weight)))
+		// disputes
+		.saturating_add((103_899_000 as Weight).saturating_mul(n_disputes as Weight))
+		.saturating_add(weights.reads(61 as Weight))
+		.saturating_add(weights.reads((1 as Weight).saturating_mul(n_disputes as Weight)))
+		.saturating_add(weights.writes(212 as Weight))
+		.saturating_add(weights.writes((2 as Weight).saturating_mul(n_disputes as Weight)))
+		// bitfields
+		.saturating_add((10_000_000 as Weight).saturating_mul(n_disputes as Weight))
+		.saturating_add(weights.reads(10 as Weight))
+		.saturating_add(weights.reads((20 as Weight).saturating_mul(n_bitfields as Weight)))
+		.saturating_add(weights.writes(10 as Weight))
+		.saturating_add(weights.writes((20 as Weight).saturating_mul(n_bitfields as Weight)))
+}
+
+/// Extract the static weight.
+fn static_weight<T: Config>() -> Weight {
+	paras_inherent_total_weight::<T>(0, 0, 0)
```

Comment:
@Lldenaurois I think if we input 0 to a benchmark that does everything we can get a good value here